godtossing/Pos/Ssc/GodTossing/Core/Core.hs

```diff
@@ -318,22 +311,26 @@ mkGtProof payload =
 stripGtPayload :: Byte -> GtPayload -> Maybe GtPayload
 stripGtPayload lim payload | biSize payload <= lim = Just payload
 stripGtPayload lim payload = case payload of
-    (CertificatesPayload vssmap) -> CertificatesPayload <$> stripHashMap lim vssmap
+    (CertificatesPayload vssmap) ->
+        CertificatesPayload <$> stripVss lim vssmap
     (CommitmentsPayload (getCommitmentsMap -> comms0) certs0) -> do
-        let certs = stripHashMap limCerts certs0
+        let certs = stripVss limCerts certs0
         let comms = stripHashMap (lim - biSize certs) comms0
         CommitmentsPayload <$> (mkCommitmentsMapUnsafe <$> comms) <*> certs
     (OpeningsPayload openings0 certs0) -> do
-        let certs = stripHashMap limCerts certs0
+        let certs = stripVss limCerts certs0
         let openings = stripHashMap (lim - biSize certs) openings0
         OpeningsPayload <$> openings <*> certs
     (SharesPayload shares0 certs0) -> do
-        let certs = stripHashMap limCerts certs0
+        let certs = stripVss limCerts certs0
         let shares = stripHashMap (lim - biSize certs) shares0
         SharesPayload <$> shares <*> certs
   where
     limCerts = lim `div` 3 -- certificates are 1/3 less important than everything else
                            -- this is a random choice in fact
+    stripVss l = fmap UnsafeVssCertificatesMap .
```

Comment:
I believe it would be good to write a comment here explaining why it's safe (it's quite obvious, but still).